Lib/test/test_pathlib.py

```diff
@@ -1520,6 +1520,12 @@ def test_resolve_common(self):
             self._check_resolve_relative(p, P(BASE, 'foo', 'in', 'spam'), False)
         # Now create absolute symlinks
         d = tempfile.mkdtemp(suffix='-dirD')
+        # bpo-32907: Canonicalize the temporary directory path since
+        # it is used in comparisons with other canonicalized paths below.
+        # Use Path.resolve() to ensure that it doesn't contain the username in
+        # "short" (8.3) format on Windows (os.path.realpath() can't be used
+        # for that).
+        d = str(P(d).resolve())
```

Comment:
Resolving makes sense in general. For example, "C:\Users" could be a junction or the user's `%LocalAppData%` may be a junction or relocated to a path that traverses a junction, or the user's `%TEMP%` could be non-standard.

That said, it might be useful in other cases to add  `nt._getlongpathname` and `nt._getshortpathname` functions that call WinAPI `GetLongPathName` and `GetShortPathName`, and add methods to get the short and long paths to pathlib's Windows flavour. In particular I can see someone wanting the long path, but not the fully resolved path.

Reply:
I agree. I've changed the wording a bit to make it look less like a Windows-specific fix. Thank you!